### ISSUE TITLE
ubuntu 18.04 - make install - miss ldconfig?

### ISSUE THREAD

**Issue by boaks** [obgm/libcoap#490]
if libcoap is build on a new system, "make install" installs the libraries at "/usr/local/lib".
But if the `coap-client` is started, it reports, that the libcoap is missing.
AFAIK, a call of ldconfig is required, but missing. After that, the `coap-client` executes without that failure.

**Comment by obgm** [comment]
True, but calling ldconfig on the install directory (or adding it to `/etc/ld.so.conf` is system configuration. IMO this should not belong into the Makefile's responsibility. @tijuca ?

**Comment by tijuca** [comment]
I guess then is something miss configured an the system.

I'm not working with Ubuntu but on the following parts Ubuntu and Debian is behaving the same.
O.k. looking at `/etc/ld.so.conf` we will see that content:
```
include /etc/ld.so.conf.d/*.conf
```
And in  `/etc/ld.so.conf.d/`  you will find a file called `libc.conf` with that content:
```
# libc default configuration
/usr/local/lib
```
Which package belongs to `/etc/ld.so.conf.d/libc.conf`?
```
$ dpkg -S /etc/ld.so.conf.d/libc.conf
libc-bin: /etc/ld.so.conf.d/libc.conf
```
But the package `libc-bin` is an essential package, means this is always installed as it is really essential. :smiley: 
```
$ apt show libc-bin | grep Essential

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Essential: yes
```
In the end ldconfig is configured as default to also respect the folder /usr/local/lib, there aren't normally any further steps needed if you install libcoap into /usr/local and did not configured any specific target folder.
This is also the first complain that using the default installation folder isn't working so far I remember.

OTOH I agree with obgm that ldconfig calls are not part of any installation step.

**Comment by obgm** [comment]
The question if `/usr/local/lib` is included somewhere in `/etc/ld.so.conf` put aside—if changing `--prefix` during configure, you would have the same issue as described by @boaks.  IIRC, the linker outputs a note explicitly stating the steps you might want to make `ld` find the shared libraries.

**Comment by boaks** [comment]
I called ldconfig in my home directory.
As far as I understood that ld-libraries-system, if you alter something with that libraries, either a "restart" or a explicit call to `ldconfig` is required.
After I execute the ldconfig, the example execs have worked.

**Comment by boaks** [comment]
> The question if /usr/local/lib is included somewhere in /etc/ld.so.conf

Yes, it was, but without refresh the libs-cache with `ldconfig`, it seems not to work. 
If wanted, I may test this with a new ubuntu 18.04 installation.
